packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -76,6 +76,8 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     triggerRef,
     contentRef,
     requestOpenChange,
+    dialogBodyID: useId('dialog-body-'),
+    dialogTitleID: useId('dialog-title-'),
```

Comment:
It appears that it's going to break once `DialogTitle` will have `id`:

```ts
<DialogTitle id="foo" />
```

- `id` from the context will not match `id` on an element

Reply:
That's true, and that's a problem for all components that have `aria-labelledby`, `aria-desribedby`, `aria-controls` and other `IDREF` ARIA attributes. We should open an issue to discuss and solve this problem overall, but so far no solution for this problem has been presented!

Reply:
Would you mind to open an issue for this problem? 🙏